cdap-app-fabric/src/main/java/io/cdap/cdap/internal/bootstrap/BootstrapService.java

```diff
@@ -97,6 +97,11 @@ protected void startUp() {
             "Bootstrapping could not complete due to interruption. It will be re-run the next time CDAP starts.");
       }
 
+      // Do not start other services if no bootstrap steps were executed.
+      if (config.getSteps().isEmpty()) {
```

Comment:
on a second thought, we can also skip this step if `BootstrapConfig` does not contain step of type `BootstrapStep.Type.LOAD_SYSTEM_ARTIFACTS`.

And instead of `noOpModule` we can add `DefaultNamespaceCreator` step in module.

Reply:
Keeping a simple solution to skip starting of other services if there were no bootstrap steps.

After fixing ProgramRuntimeService and RunRecordMonitoring service, the handlers and services between appfabric service and processor. Then the wiring for bootstrapping will change.

Reply:
one question, if `NamespaceHttpHandler` & `ProfileHttpHandler` both are in `app-fabric service` used to create namespaces & compute profiles, why do we want to keep the steps to create `Default Namespace` & `Native Compute Profile` in `app-fabric processor`? I am still unable to understand this part.

Reply:
As discussed offline:

1. Appfabric server can be run as k8s deployment in distributed mode where it will not run StorageMain (initContainer). So we should not try to perform steps for creating namespace or load system artifacts as can fail if storage tables are not initialized.
2. To improve the solution, I have moved the condition at the top to return early if no config steps are mentioned.
3. Appfabric processor executes bootsrap steps and runs the other necessary services during start.